### ISSUE TITLE
Create a cloudflare_zone_settings_ssl module

### ISSUE THREAD

**Issue by AndrewJDawes** [codekaizen-github/cloudflare-ansible-collection#9]
In Cloudflare dashboard UI, we can change settings for a Zone. One of the settings we can change is the SSL setting that is used for that Zone.

![https://img001.prntscr.com/file/img001/I1isrnU-RYywhKAStV04IA.png](https://img001.prntscr.com/file/img001/I1isrnU-RYywhKAStV04IA.png)

The Cloudflare API exposes access to this SSL setting (and other settings). See the [docs](https://developers.cloudflare.com/api/operations/zone-settings-get-all-zone-settings).

We would like an Ansible module that will allow us to fetch (`state: fetched`), create (`state: present`), and update (`state: present`) the `ssl` setting for a Zone using the Cloudflare API.

Here is an example (using `curl`) of how you can currently use the Cloudflare API to get settings.

```bash
# List your zones. Use the `name` query param to retrieve your zone by name.
curl --request GET \
  --url https://api.cloudflare.com/client/v4/zones?name={ YOUR_DOMAIN_NAME } \
  --header 'Authorization: Bearer { YOUR_TOKEN_HERE }' \
  --header 'Content-Type: application/json'

# List all the Zone's settings (not needed if only wanting to edit `ssl`, but helpful to visualize). Use the Zone ID from the previous request.
curl --request GET \
  --url https://api.cloudflare.com/client/v4/zones/{ YOUR_ZONE_ID_HERE }/settings \
  --header 'Authorization: Bearer { YOUR_TOKEN_HERE }' \
  --header 'Content-Type: application/json'

# Get the Zone's SSL setting
curl --request GET \
  --url https://api.cloudflare.com/client/v4/zones/{ YOUR_ZONE_ID_HERE }/settings/ssl \
  --header 'Authorization: Bearer { YOUR_TOKEN_HERE }' \
  --header 'Content-Type: application/json'

# Update the Zone's SSL setting to `full`.
curl --request PATCH \
  --url https://api.cloudflare.com/client/v4/zones/{ YOUR_ZONE_ID_HERE }/settings/ssl \
  --header 'Authorization: Bearer { YOUR_TOKEN_HERE }' \
  --header 'Content-Type: application/json' \
  --data '{
  "value": "full"
}'
```

- [x] Read through the existing `community.general.cloudflare_dns` [module's code](https://github.com/ansible-collections/community.general/blob/6e0142fe3aa7c0e8087a51e352c931dc7ddcf7f0/plugins/modules/cloudflare_dns.py#L1) to understand how they handle authentication to Cloudflare's API
- [x] Manually create a Cloudflare account
- [x] Manually create a Cloudflare zone
- [x] Create an [API token](https://dash.cloudflare.com/profile/api-tokens) in Cloudflare that has access to `Edit` that Zone
- [x] Create an Ansible module inside this collection named cloudflare_zone_settings_ssl
- [x] It should accept these params at a minimum: `zone` (name, not ID), `api_token`, `state` (one of `fetched|present`), `value` (one of `full|flexible|strict|off`.
- [x] It should use Cloudflare's REST API to fetch or change the SSL settings on a Zone.
- [x] It should not use `curl` - please try to use `ansible.module_utils.urls.fetch_url` first, and then Python's [Requests](https://pypi.org/project/requests/) package if that doesn't pan out.

Thanks!

**Comment by AndrewJDawes** [comment]
@kwyland22 - hoping you could take on this one! Please let me know if you have any questions. Feel free to drop them directly in Slack!

**Comment by kwyland22** [comment]
@AndrewJDawes I have completed this in the "cf_zone_settings_ssl" branch! I opened up a pull request. Let me know what you think!

**Comment by AndrewJDawes** [comment]
Closed by #10
